### PR TITLE
Merge the debian patches for 3.7

### LINKED ISSUE #1682: Apply debian patches upstream (3.7 edition)
As per personal correspondence with Maitland, there's quite a few patches that need merging into 3.7.

This is the issue to track myself doing that.

### PR DESCRIPTION
**Pull request by marcusmueller**
Closes #1682.

### CODE REVIEW

File: gr-utils/python/modtool/gr-newmod/docs/doxygen/Doxyfile.swig_doc.in [hunk 1]
```diff
@@ -121,7 +121,7 @@ INLINE_INHERITED_MEMB  = NO
 # path before files name in the file list and in the header files. If set
 # to NO the shortest path that makes the file name unique will be used.
 
-FULL_PATH_NAMES        = YES
+FULL_PATH_NAMES        = NO
```
Comment (noc0lour) [inline gr-utils/python/modtool/gr-newmod/docs/doxygen/Doxyfile.swig_doc.in]: What's the result of this change?

File: gnuradio-runtime/lib/thread/thread.cc [hunk 2]
```diff
@@ -166,7 +166,7 @@ namespace gr {
 
 
 #elif defined(macintosh) || defined(__APPLE__) || defined(__APPLE_CC__) || \
-    defined(__FreeBSD__)
+  defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__GNU__)
```
Comment (noc0lour) [inline gnuradio-runtime/lib/thread/thread.cc]: not sure why `__GNU__` is in here and why there is an `#else` case below. I think `__GNU__` is intended to go into the `#else` case.

File: gr-audio/lib/CMakeLists.txt [hunk 3]
```diff
@@ -44,7 +44,7 @@ endif(ENABLE_GR_CTRLPORT)
 ########################################################################
 find_package(ALSA)
 
-if(ALSA_FOUND)
+if((ALSA_FOUND) AND NOT (CMAKE_SYSTEM_NAME STREQUAL "kFreeBSD") AND NOT (CMAKE_SYSTEM_NAME STREQUAL GNU))
```
Comment (noc0lour) [inline gr-audio/lib/CMakeLists.txt]: Not sure about the GNU part again? Is this correct? Will this disable ALSA on all GNU platforms?

File: gr-fcd/CMakeLists.txt [hunk 4]
```diff
@@ -31,7 +31,7 @@ endif()
 include(GrComponent)
 
 # FIXME make better conditionalization
```
Comment (noc0lour) [inline gr-fcd/CMakeLists.txt]: Also remove this stray comment then

File: gr-utils/python/modtool/gr-newmod/docs/doxygen/Doxyfile.swig_doc.in (hunk 1 above)
Comment (marcusmueller) [reply gr-utils/python/modtool/gr-newmod/docs/doxygen/Doxyfile.swig_doc.in]: That the absolute build path doesn't end up in docs, meaning the built files are not dependent on where they were built.

File: gnuradio-runtime/lib/thread/thread.cc (hunk 2 above)
Comment (marcusmueller) [reply gnuradio-runtime/lib/thread/thread.cc]: AFAIK, that's GNU Hurd: https://sourceforge.net/p/predef/wiki/OperatingSystems/

File: gr-audio/lib/CMakeLists.txt (hunk 3 above)
Comment (marcusmueller) [reply gr-audio/lib/CMakeLists.txt]: No, it will not build ALSA blocks on GNU Hurd

File: gr-fcd/CMakeLists.txt (hunk 4 above)
Comment (marcusmueller) [reply gr-fcd/CMakeLists.txt]: yep, will do.

File: gr-audio/lib/CMakeLists.txt (hunk 3 above)
Comment (noc0lour) [reply gr-audio/lib/CMakeLists.txt]: Ah, then only Linux, MacOS and "normal" FreeBSD will build ALSA blocks. Why not kFreeBSD & GNU Hurd if ALSA is available?

File: gnuradio-runtime/lib/thread/thread.cc (hunk 2 above)
Comment (noc0lour) [reply gnuradio-runtime/lib/thread/thread.cc]: Is fine then.

File: gr-audio/lib/CMakeLists.txt (hunk 3 above)
Comment (marcusmueller) [reply gr-audio/lib/CMakeLists.txt]: AFAIK GNU Hurd simply has no proper ALSA: see this *exquisite* documentation [or chat log](https://www.gnu.org/software/hurd/open_issues/anatomy_of_a_hurd_system.html) and debian's own guideline to disable ALSA on hurd: https://www.debian.org/ports/hurd/hurd-devel-debian. 

All in all, it's absolutely not surprising that Hurd doesn't do monolithic syscall-style ALSA. It's a microkernel! Of course,it emulates OSS, but ALSA's simply not there.

kFreeBSD simply doesn't have an ALSA interface. There's salsa to (very imperfectly) emulate ALSA, afaik.

So, to answer the question "why not ALSA if ALSA is available?": It's not available.

**Comment by marcusmueller** [comment]
yay! passing tests, @noc0lour, should we wrap this up?

**Comment by noc0lour** [comment]
You got lucky, and yes I think this is good to go.

**Comment by marcusmueller** [comment]
The question that really irks me is whether "rebase and merge" autosquashes --fixup commits or not – but I'm not willing to test that on master.

**Pull request merged**